### ISSUE TITLE
Assembly browser back/forward navigation opens new Assembly Browser window

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#7542]
Build Information
Release ID: 801002446
Git revision: de79902bc8aa4a23eb717e319c33e94c07033d83
Build date: 2019-05-10 00:06:52+00
Build branch: release-8.1
Xamarin extensions: a79ff61f160bf08b56a8b36bee4fdfc276800b8d

VS Mac 8.0 seems OK.

Repro:

1. Create a C# project.
2. Expand References and double click System to open Assembly Browser.
3. Click different assemblies in the Assembly Browser.
4. Navigate back using the < button - this goes back to MyClass most of the time but some times this opens a new Assembly Browser.
5. Navigate forward - seems to open a new Assembly Browser window.
6. Navigate forward - opens another assembly browser window.

![AssemblyBrowserNavigation](https://user-images.githubusercontent.com/372361/57520354-5b03e480-7315-11e9-80a3-09071299094b.gif)

> VS bug [#891266](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/891266)

**Comment by mrward** [comment]
I think that this opening of other assembly viewers might be the cause of a fatal crash which you get sometimes clicking a type in the assembly viewer to go to that type.

```
ERROR [2019-05-10 11:27:44Z]: An unhandled exception has occurred. Terminating Visual Studio? False
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.AssemblyBrowser.AssemblyBrowserWidget+<>c__DisplayClass72_1.<EnsureDefinitionsLoaded>b__0 (System.Object o, System.EventArgs args) [0x00050] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MonoDevelop.AssemblyBrowser/MonoDevelop.AssemblyBrowser/AssemblyBrowserWidget.cs:1265 
  at Gtk.Application+InvokeProxy.HandlerInternal (System.IntPtr data) [0x0001d] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/gtk-sharp-None/gtk/Application.cs:199 
ERROR [2019-05-10 11:27:44Z]: An unhandled exception has occurred. Terminating Visual Studio? False
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.AssemblyBrowser.AssemblyBrowserWidget+<>c__DisplayClass72_1.<EnsureDefinitionsLoaded>b__0 (System.Object o, System.EventArgs args) [0x00050] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MonoDevelop.AssemblyBrowser/MonoDevelop.AssemblyBrowser/AssemblyBrowserWidget.cs:1265 
  at Gtk.Application+InvokeProxy.HandlerInternal (System.IntPtr data) [0x0001d] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/gtk-sharp-None/gtk/Application.cs:199 
ERROR [2019-05-10 11:27:48Z]: An unhandled exception has occurred. Terminating Visual Studio? False
System.ArgumentException: The tasks argument included a null value.
Parameter name: tasks
  at System.Threading.Tasks.Task.WhenAll[TResult] (System.Collections.Generic.IEnumerable`1[T] tasks) [0x000a2] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corert/src/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:5691 
  at MonoDevelop.AssemblyBrowser.AssemblyBrowserWidget.Open (System.String url, MonoDevelop.AssemblyBrowser.AssemblyLoader currentAssembly, System.Boolean expandNode) [0x00022] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MonoDevelop.AssemblyBrowser/MonoDevelop.AssemblyBrowser/AssemblyBrowserWidget.cs:913 
  at MonoDevelop.AssemblyBrowser.AssemblyBrowserViewContent.Open (System.String documentationCommentId, System.Boolean openInPublicOnlyMode, System.Boolean expandNode) [0x0000f] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MonoDevelop.AssemblyBrowser/MonoDevelop.AssemblyBrowser/AssemblyBrowserViewContent.cs:149 
  at MonoDevelop.AssemblyBrowser.AssemblyBrowserNavigationPoint.DoShow () [0x000b2] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MonoDevelop.AssemblyBrowser/MonoDevelop.AssemblyBrowser/AssemblyBrowserNavigationPoint.cs:56 
FATAL ERROR [2019-05-10 11:27:51Z]: An unhandled exception has occurred. Terminating Visual Studio? True
System.ArgumentException: The tasks argument included a null value.
Parameter name: tasks
  at System.Threading.Tasks.Task.WhenAll[TResult] (System.Collections.Generic.IEnumerable`1[T] tasks) [0x000a2] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corert/src/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:5691 
  at MonoDevelop.AssemblyBrowser.AssemblyBrowserWidget.Open (System.String url, MonoDevelop.AssemblyBrowser.AssemblyLoader currentAssembly, System.Boolean expandNode) [0x00022] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MonoDevelop.AssemblyBrowser/MonoDevelop.AssemblyBrowser/AssemblyBrowserWidget.cs:913 
  at MonoDevelop.AssemblyBrowser.AssemblyBrowserWidget+<>c__DisplayClass51_0.<SetReferencedSegments>b__0 (MonoDevelop.Ide.Editor.LinkRequest request) [0x000e6] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MonoDevelop.AssemblyBrowser/MonoDevelop.AssemblyBrowser/AssemblyBrowserWidget.cs:856 
  at MonoDevelop.SourceEditor.LinkMarker.Mono.TextEditor.IActionTextLineMarker.MouseReleased (Mono.TextEditor.MonoTextEditor editor, Mono.TextEditor.MarginMouseEventArgs args) [0x0003a] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/TextMarker/LinkMarker.cs:73 
  at Mono.TextEditor.TextViewMargin.MouseReleased (Mono.TextEditor.MarginMouseEventArgs args) [0x000e4] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/TextViewMargin.cs:2490 
  at Mono.TextEditor.TextArea.OnButtonReleaseEvent (Gdk.EventButton e) [0x0002a] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/TextArea.cs:1503 
  at Gtk.Widget.buttonreleaseevent_cb (System.IntPtr widget, System.IntPtr evnt) [0x00016] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/gtk-sharp-None/gtk/generated/Widget.cs:2169 
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Exception: We were unable to start the Microsoft Error Reporting client.
  at (wrapper managed-to-native) Mono.Runtime.SendMicrosoftTelemetry_internal(intptr,ulong,ulong)
  at Mono.Runtime.SendMicrosoftTelemetry (System.String payload_str, System.UInt64 portable_hash, System.UInt64 unportable_hash) [0x00013] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/mcs/class/corlib/Mono/Runtime.cs:128 
  at Mono.Runtime.SendExceptionToTelemetry (System.Exception exc) [0x0000a] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/mcs/class/corlib/Mono/Runtime.cs:147 
  at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke(System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0003b] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Reflection/MonoMethod.cs:305 
   --- End of inner exception stack trace ---
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0004c] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Reflection/MonoMethod.cs:313 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/reflection/methodbase.cs:229 
  at Xamarin.IdeCustomization.XamarinCustomizer.CallMonoMethod (System.Type monoType, System.String methodName, System.Object[] parameters) [0x00015] in /Users/vsts/agent/2.150.3/work/1/s/md-addins/Xamarin.Ide/Xamarin.IdeCustomization/Xamarin.IdeCustomization/XamarinCustomizer.cs:86 
  at Xamarin.IdeCustomization.XamarinCustomizer.SendExceptionToTelemetry (System.Exception exception) [0x0002e] in /Users/vsts/agent/2.150.3/work/1/s/md-addins/Xamarin.Ide/Xamarin.IdeCustomization/Xamarin.IdeCustomization/XamarinCustomizer.cs:210 
  at Xamarin.Ide.LogReporting.CounterMapping+ExceptionCounterMap.ProcessCounter (MonoDevelop.Core.Instrumentation.Counter counter, MonoDevelop.Core.Instrumentation.CounterValue value) [0x0001f] in /Users/vsts/agent/2.150.3/work/1/s/md-addins/Xamarin.Ide/Xamarin.IdeCustomization/Xamarin.Ide.LogReporting/CounterMapping.cs:608 
  at Xamarin.Ide.LogReporting.CounterMapping.Process (MonoDevelop.Core.Instrumentation.Counter counter, MonoDevelop.Core.Instrumentation.CounterValue value) [0x00014] in /Users/vsts/agent/2.150.3/work/1/s/md-addins/Xamarin.Ide/Xamarin.IdeCustomization/Xamarin.Ide.LogReporting/CounterMapping.cs:365 
  at Xamarin.Ide.LogReporting.TelemetryService.ReportCounter (MonoDevelop.Core.Instrumentation.Counter counter, MonoDevelop.Core.Instrumentation.CounterValue value) [0x00000] in /Users/vsts/agent/2.150.3/work/1/s/md-addins/Xamarin.Ide/Xamarin.IdeCustomization/Xamarin.Ide.LogReporting/TelemetryService.cs:280 
  at Xamarin.Ide.LogReporting.TelemetryInstrumentationConsumer.ConsumeValue (MonoDevelop.Core.Instrumentation.Counter counter, MonoDevelop.Core.Instrumentation.CounterValue value) [0x00000] in /Users/vsts/agent/2.150.3/work/1/s/md-addins/Xamarin.Ide/Xamarin.IdeCustomization/Xamarin.Ide.LogReporting/TelemetryInstrumentationConsumer.cs:24 
  at MonoDevelop.Core.Instrumentation.Counter.StoreValue (System.String message, MonoDevelop.Core.Instrumentation.ITimeCounter timer, System.Collections.Generic.IDictionary`2[TKey,TValue] metadata) [0x000f3] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core.Instrumentation/Counter.cs:227 
  at MonoDevelop.Core.Instrumentation.Counter.Inc (System.Int32 n, System.String message, System.Collections.Generic.IDictionary`2[TKey,TValue] metadata) [0x00035] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core.Instrumentation/Counter.cs:294 
  at MonoDevelop.Core.Instrumentation.Counter`1[T].Inc (T metadata) [0x00000] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core.Instrumentation/Counter.cs:417 
  at (wrapper remoting-invoke-with-check) MonoDevelop.Core.Instrumentation.Counter`1[MonoDevelop.Ide.UnhandledExceptionMetadata].Inc(MonoDevelop.Ide.UnhandledExceptionMetadata)
  at MonoDevelop.Ide.IdeStartup.HandleException (System.Exception ex, System.Boolean willShutdown) [0x00066] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs:679 
  at MonoDevelop.Ide.IdeStartup+<>c.<SetupExceptionManager>b__28_1 (GLib.UnhandledExceptionArgs args) [0x00000] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs:648 
  at GLib.ExceptionManager.RaiseUnhandledException (System.Exception e, System.Boolean is_terminal) [0x0005b] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/gtk-sharp-None/glib/ExceptionManager.cs:64 
```

**Comment by mkrueger** [comment]
It's caused by navigation point changes
var result = await Ide.IdeApp.Workbench.OpenDocument (AssemblyBrowserDescriptor.Instance);

in navigation point always opens a new assembly browser instead of re-using the open one.

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-14] Comment by **xamarinc**



Fixed in version 8.3.0.107 (master)



Pull Request #7684 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 8a2cc0a97d7533a56843ad1ccb14969713465207 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-15] Comment by **xamarinc**



Fixed in version 8.1.0.2603 (release-8.1)



Pull Request #7688 merged by: Lluis Sanchez  
 Author: mono  
 Commit: f20c0fb05d18110746cbaad2f6d8b2547b89f31d (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-22] Comment by **xamarinc**



Fixed in version 8.2.0.542 (release-8.2)



Pull Request #7684 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 8a2cc0a97d7533a56843ad1ccb14969713465207 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-24] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #7684 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 8a2cc0a97d7533a56843ad1ccb14969713465207 (xamarin/monodevelop)
